### PR TITLE
Feature/aw1 aeas

### PR DESCRIPTION
**Pull request by DavidMinarsch**
## Proposed changes

This PR adds AEAs for AW1.

## Fixes

-

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have checked that the documentation about the `aea cli` tool works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

-

### CODE REVIEW

File: packages/fetchai/agents/confirmation_aea_aw1/aea-config.yaml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+protocols:
+- fetchai/contract_api:0.7.0
+- fetchai/default:0.8.0
+- fetchai/ledger_api:0.6.0
+- fetchai/oef_search:0.9.0
+- fetchai/register:0.1.0
+- fetchai/signing:0.6.0
+- fetchai/state_update:0.6.0
+skills:
+- fetchai/confirmation_aw1:0.1.0
```
Comment (5A11) [inline packages/fetchai/agents/confirmation_aea_aw1/aea-config.yaml:23]: You mentioned `fetchai/simple_service_registration` in the readme but it does not appear in this.

File: packages/fetchai/protocols/register/README.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    info: pt:dict[pt:str, pt:str]
+...
+---
+initiation: [register]
+reply:
+  register: [success, error]
+  success: []
+  error: []
+termination: [success, error]
+roles: {agent, agent}
```
Comment (5A11) [inline packages/fetchai/protocols/register/README.md:34]: When the roles are the same, you only need to insert one:

```
roles: {agent}
```

File: packages/fetchai/skills/confirmation_aw1/skill.yaml [hunk 3]
```diff
@@ -0,0 +1,47 @@
+name: confirmation_aw1
+author: fetchai
+version: 0.1.0
+type: skill
+description: The scaffold skill is a scaffold for your own skill implementation.
```
Comment (5A11) [inline packages/fetchai/skills/confirmation_aw1/skill.yaml:5]: Needs updating

File: packages/fetchai/skills/registration_aw1/handlers.py [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+        """
+        register_msg = cast(RegisterMessage, message)
+
+        # recover dialogue
+        register_dialogues = cast(RegisterDialogues, self.context.register_dialogues)
+        register_dialogue = cast(
+            Optional[RegisterDialogue], register_dialogues.update(register_msg)
+        )
+        if register_dialogue is None:
+            self._handle_unidentified_dialogue(register_msg)
```
Comment (5A11) [inline packages/fetchai/skills/registration_aw1/handlers.py:66]: Quality of life change:
Missing some logging info here and everywhere else 

(but not that high priority to get things going)

File: packages/fetchai/skills/registration_aw1/skill.yaml [hunk 5]
```diff
@@ -0,0 +1,48 @@
+name: registration_aw1
+author: fetchai
+version: 0.1.0
+type: skill
+description: The scaffold skill is a scaffold for your own skill implementation.
```
Comment (5A11) [inline packages/fetchai/skills/registration_aw1/skill.yaml:5]: Incorrect description

**Review by 5A11 — APPROVED** [review]
See comments

File: packages/fetchai/agents/confirmation_aea_aw1/aea-config.yaml (hunk 1 above)
Comment (DavidMinarsch) [reply packages/fetchai/agents/confirmation_aea_aw1/aea-config.yaml:23]: I added it now, thanks

File: packages/fetchai/protocols/register/README.md (hunk 2 above)
Comment (DavidMinarsch) [reply packages/fetchai/protocols/register/README.md:34]: Yes correct, thanks

File: packages/fetchai/skills/confirmation_aw1/skill.yaml (hunk 3 above)
Comment (DavidMinarsch) [reply packages/fetchai/skills/confirmation_aw1/skill.yaml:5]: thanks, will do

File: packages/fetchai/skills/registration_aw1/handlers.py (hunk 4 above)
Comment (DavidMinarsch) [reply packages/fetchai/skills/registration_aw1/handlers.py:66]: yep can be done but as you say, low priority atm

File: packages/fetchai/skills/registration_aw1/skill.yaml (hunk 5 above)
Comment (DavidMinarsch) [reply packages/fetchai/skills/registration_aw1/skill.yaml:5]: thanks, will update

**Comment by 5A11** [comment]
lgtm

**Comment by marcofavorito** [comment]
The server shouldn't refund more than one time the same `registration_aea_aw1`.

File: packages/fetchai/skills/confirmation_aw1/registration_db.py [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+
+    def _initialise_backend(self) -> None:
+        """Set up database and initialise the tables."""
+        if os.path.isfile(self.db_path):
+            return
+        self._execute_single_sql(
+            "CREATE TABLE IF NOT EXISTS registered_table (address TEXT, ethereum_address TEXT, "
+            "ethereum_signature TEXT, fetchai_signature TEXT, "
+            "developer_handle TEXT, tweet TEXT)"
+        )
```
Comment (marcofavorito) [inline packages/fetchai/skills/confirmation_aw1/registration_db.py:53]: you could consider using some field as primary key, so it's guaranteed unique insertion for that field.

https://www.sqlitetutorial.net/sqlite-primary-key/

File: packages/fetchai/skills/confirmation_aw1/handlers.py [hunk 7]
```diff
@@ -118,6 +118,9 @@ def _handle_register(
             register_msg.info, register_msg.sender
         )
         if is_valid:
+            strategy.lock_registration_temporarily(
+                register_msg.sender, register_msg.info
+            )
```
Comment (marcofavorito) [inline packages/fetchai/skills/confirmation_aw1/handlers.py:123]: how the lock is restored when something wrong happen? E.g. I expected a try-except-finally pattern

Comment (DavidMinarsch) [reply packages/fetchai/skills/confirmation_aw1/handlers.py:123]: we now unlock on error. if an error happens in the ledger connection we don't handle it, it stays locked. ok for this simple agent.

**Pull request merged**